Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.cpp

```diff
@@ -287,6 +288,62 @@ WASM_IPINT_EXTERN_CPP_DECL(epilogue_osr, CallFrame* callFrame)
 #endif
 #endif
 
+WASM_IPINT_EXTERN_CPP_DECL(retrieve_and_clear_exception, v128_t* stackPointer)
+{
+    VM& vm = instance->vm();
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
+    RELEASE_ASSERT(!!throwScope.exception());
+
+    if (stackPointer) {
+        // We only have a stack pointer if we're doing a catch not a catch_all
+        Exception* exception = throwScope.exception();
+        auto* wasmException = jsSecureCast<JSWebAssemblyException*>(exception->value());
+
+        ASSERT(wasmException->payload().size() == wasmException->tag().parameterCount());
+        uint64_t size = wasmException->payload().size();
+
+        for (unsigned i = 0; i < size; ++i)
+            stackPointer[size - i - 1] = { wasmException->payload()[i], wasmException->payload()[i] };
```

Comment:
Can one of these be 0 or BADBEEF?

Reply:
I'll make it 0 in release and 0x1BADBEEF in debug